### PR TITLE
Add merchant admin functionality including viewing items and creating, editing, and deleting an item

### PR DESCRIPTION
**Pull request by grwthomps**

### CODE REVIEW

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+  before_action :require_merchant_admin, except: [:index]
+
   def index
     @merchant = current_user.merchant
   end
+
+  def update
+    @item = Item.find(params[:id])
+    if request.query_string == 'toggle_status'
+      @item.toggle! :active?
```
Comment (zacisaacson) [inline app/controllers/merchant/items_controller.rb:11]: I like that query string

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+      redirect_to '/merchant/items'
+    else
+      @item.update(item_params)
+      if @item.save
+        flash.now[:success] = ["#{@item.name} has been successfully updated"]
+        redirect_to '/merchant/items'
+      else
+        flash.now[:error] = @item.errors.full_messages
+        render :edit
+      end
```
Comment (zacisaacson) [inline app/controllers/merchant/items_controller.rb:26]: Could we clean this up at all?

File: app/views/merchant/items/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<h1 class="mt-4 mb-2">Create New Item For <%=link_to @merchant.name, "/merchants/#{@merchant.id}" %></h1>
+
+<%= render partial: 'partials/item_form', locals:
```
Comment (zacisaacson) [inline app/views/merchant/items/new.html.erb:3]: Nice partial

File: app/views/merchant/items/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  {
+    path: "/merchant/items/#{@item.id}/?update",
+    method: :patch,
+    name: @item.name,
+    description: @item.description,
+    price: @item.price,
+    image: @item.image,
+    inventory: @item.inventory,
+    button_text: 'Update Item'
+  } %>
```
Comment (johnktravers) [inline app/views/merchant/items/edit.html.erb:13]: Nice use of partials

File: db/migrate/20190817190207_create_items.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@ def change
       t.string :name
       t.string :description
       t.float :price
-      t.string :image
+      t.string :image, default: 'https://i.ytimg.com/vi/Xw1C5T-fH2Y/maxresdefault.jpg'
```
Comment (johnktravers) [inline db/migrate/20190817190207_create_items.rb:7]: I love the default image

**Review by zacisaacson — APPROVED** [review]

File: spec/features/merchant_admins/item_index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    expect(page).to_not have_link(@dog_bone.name)
+    expect(page).to_not have_content(@dog_bone.description)
+    expect(page).to_not have_content("Price: $#{@dog_bone.price}")
+    expect(page).to_not have_content("Inventory: #{@dog_bone.inventory}")
+    expect(page).to_not have_css("img[src*='#{@dog_bone.image}']")
+
+    expect(page).to have_content("Dog Bone has been successfully deleted")
+  end
+end
```
Comment (johnktravers) [inline spec/features/merchant_admins/item_index_spec.rb:94]: Very thorough testing!

**Review by johnktravers — APPROVED** [review]
Looks great!

**Pull request merged**